packages/core/contracts/oracle/implementation/OptimisticOracle.sol

```diff
@@ -309,6 +348,8 @@ contract OptimisticOracle is OptimisticOracleInterface, Testable, Lockable {
             address(request.currency)
         );
 
+        // End the re-entrancy guard early to allow the caller to potentially take OO-related actions inside this callback.
+        _notEntered = true;
```

Comment:
How did you catch this bug? This is a good find

Reply:
It wasn't a bug, we just didn't allow you to call back into this contract.